metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -0,0 +1,64 @@
+package scala.meta.internal.metals
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.util.control.NonFatal
+
+import coursierapi.error.SimpleResolutionError
+
+class MtagsResolver {
+
+  private val states =
+    new ConcurrentHashMap[String, Either[Long, MtagsBinaries]]()
```

Comment:
I'll appreciate some comment, e.g. `Contains successfully resolved mtags binary or timestamp of last failed result`